plugins/napatech/source-napatech.c

```diff
@@ -171,112 +137,107 @@ static TmEcode NapatechStreamDeInit(void)
 /**
  * \brief Register the Napatech  receiver (reader) module.
  */
-void TmModuleNapatechStreamRegister(void)
+void TmModuleReceiveNapatechRegister(int slot)
 {
-    tmm_modules[TMM_RECEIVENAPATECH].name = "NapatechStream";
-    tmm_modules[TMM_RECEIVENAPATECH].ThreadInit = NapatechStreamThreadInit;
-    tmm_modules[TMM_RECEIVENAPATECH].Func = NULL;
-    tmm_modules[TMM_RECEIVENAPATECH].PktAcqLoop = NapatechPacketLoop;
-    tmm_modules[TMM_RECEIVENAPATECH].PktAcqBreakLoop = NULL;
-    tmm_modules[TMM_RECEIVENAPATECH].ThreadExitPrintStats = NapatechStreamThreadExitStats;
-    tmm_modules[TMM_RECEIVENAPATECH].ThreadDeinit = NapatechStreamThreadDeinit;
-    tmm_modules[TMM_RECEIVENAPATECH].cap_flags = SC_CAP_NET_RAW;
-    tmm_modules[TMM_RECEIVENAPATECH].flags = TM_FLAG_RECEIVE_TM;
-    tmm_modules[TMM_RECEIVENAPATECH].Init = NapatechStreamInit;
-    tmm_modules[TMM_RECEIVENAPATECH].DeInit = NapatechStreamDeInit;
+    tmm_modules[slot].name = "NapatechStream";
+    tmm_modules[slot].ThreadInit = NapatechStreamThreadInit;
+    tmm_modules[slot].Func = NULL;
+    tmm_modules[slot].PktAcqLoop = NapatechPacketLoop;
+    tmm_modules[slot].PktAcqBreakLoop = NULL;
+    tmm_modules[slot].ThreadExitPrintStats = NapatechStreamThreadExitStats;
+    tmm_modules[slot].ThreadDeinit = NapatechStreamThreadDeinit;
+    tmm_modules[slot].cap_flags = SC_CAP_NET_RAW;
+    tmm_modules[slot].flags = TM_FLAG_RECEIVE_TM;
+    tmm_modules[slot].Init = NapatechStreamInit;
+    tmm_modules[slot].DeInit = NapatechStreamDeInit;
 }
 
 /**
  * \brief Register the Napatech decoder module.
  */
-void TmModuleNapatechDecodeRegister(void)
+void TmModuleDecodeNapatechRegister(int slot)
 {
-    tmm_modules[TMM_DECODENAPATECH].name = "NapatechDecode";
-    tmm_modules[TMM_DECODENAPATECH].ThreadInit = NapatechDecodeThreadInit;
-    tmm_modules[TMM_DECODENAPATECH].Func = NapatechDecode;
-    tmm_modules[TMM_DECODENAPATECH].ThreadExitPrintStats = NULL;
-    tmm_modules[TMM_DECODENAPATECH].ThreadDeinit = NapatechDecodeThreadDeinit;
-    tmm_modules[TMM_DECODENAPATECH].cap_flags = 0;
-    tmm_modules[TMM_DECODENAPATECH].flags = TM_FLAG_DECODE_TM;
+    tmm_modules[slot].name = "DecodeNapatech";
```

Comment:
Is this done in name only to match pfring?

Reply:
The name is for lookup purposes only so `TmModuleGetByName` can locate the plugin.